Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -312,18 +314,26 @@ void MediaPlayerPrivateGStreamerMSE::sourceSetup(GstElement* sourceElement)
 
 void MediaPlayerPrivateGStreamerMSE::updateStates()
 {
+    GstState currentState, pending;
+    GstStateChangeReturn change = gst_element_get_state(m_pipeline.get(), &currentState, &pending, 0);
+
+    bool isSeeking = m_isSeeking && change == GST_STATE_CHANGE_ASYNC && currentState == GST_STATE_PAUSED && pending == GST_STATE_PAUSED;
     bool shouldBePlaying = (!m_isPaused && readyState() >= MediaPlayer::ReadyState::HaveFutureData && m_playbackRatePausedState != PlaybackRatePausedState::RatePaused)
         || m_playbackRatePausedState == PlaybackRatePausedState::ShouldMoveToPlaying;
-    GST_DEBUG_OBJECT(pipeline(), "shouldBePlaying = %s, m_isPipelinePlaying = %s", boolForPrinting(shouldBePlaying), boolForPrinting(m_isPipelinePlaying));
-    if (shouldBePlaying && !m_isPipelinePlaying) {
+    GST_DEBUG_OBJECT(pipeline(), "shouldBePlaying = %s, m_isPipelinePlaying = %s, is seeking %s", boolForPrinting(shouldBePlaying),
+        boolForPrinting(m_isPipelinePlaying), boolForPrinting(isSeeking));
+    if (!isSeeking && shouldBePlaying && !m_isPipelinePlaying) {
         if (!changePipelineState(GST_STATE_PLAYING))
-            GST_ERROR_OBJECT(pipeline(), "Setting the pipeline to PLAYING failed");
-        m_isPipelinePlaying = true;
-        m_playbackRatePausedState = PlaybackRatePausedState::Playing;
-    } else if (!shouldBePlaying && m_isPipelinePlaying) {
+            GST_DEBUG_OBJECT(pipeline(), "Setting the pipeline to PLAYING could not suceed right now");
```

Comment:
In what cases?

Reply:
Well, you can read that in the code yourself but I'll mention that in addition to failures, changing the pipeline while seeking will reject the change now as well. Pipeline will be moved to playing later after finishing the seek.